stake-pool/js/src/test.ts

```diff
@@ -0,0 +1,194 @@
+        }),
+      });
+
+      for (const key in decodedData) {
+        if (expectedData[key] instanceof BN) {
+          assert.ok(expectedData[key].eq(decodedData[key]));
+        } else {
+          for (const subkey in decodedData[key]) {
+            if (decodedData[key][subkey] instanceof BN) {
+              assert.ok(decodedData[key][subkey].eq(expectedData[key][subkey]));
+            } else {
+              assert.deepStrictEqual(
+                expectedData[key][subkey],
+                decodedData[key][subkey],
+              );
+            }
+          }
+        }
+      }
+    });
+  });
+
+  describe('ValidatorListAccount', () => {
+    it('should successfully decode ValidatorListAccount account data', () => {
+      const decodedData = schema.ValidatorList.decodeUnchecked(
+        new Buffer(
+          '020a0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
+          'hex',
+        ),
+      );
+      const expectedData = new schema.ValidatorList({
+        accountType: new schema.AccountType({
+          ValidatorList: new schema.AccountTypeEnum({}),
+        }),
+        maxValidators: 10,
+        validators: [],
+      });
+
+      assert.deepStrictEqual(decodedData, expectedData);
+    });
+
+    it('should successfully decode ValidatorListAccount with nonempty ValidatorInfo', () => {
+      // TODO also test for decoding ValidatorListAccount with actual ValidatorInfo
+      // Do this once we have a stake pool with validators deployed on testnet
+      assert.equal(true, true);
+    });
+  });
+});
+
+describe('index.ts/PrettyPrintPubkey', () => {
+  it('should successfully pretty print a pubkey', () => {
+    assert.equal(
+      index.prettyPrintPubKey(
+        new schema.PublicKey({
+          value: new BN(
+            '99572085579321386496717000324290408927851378839748241098946587626478579848783',
+          ),
+        }),
+      ),
+      '6MfzrQUzB2mozveRWU9a77zMoQzSrYa4Gq46KswjupQB',
+    );
+  });
+});
+
+describe('Integration test', () => {
```

Comment:
We can keep this for now, but when this package is integrated in the CI, we should move these tests into a different place as a separate suite of tests.  This way, we'll be sure to never break backwards compatibility :smile:

Reply:
done in https://github.com/solana-labs/solana-program-library/pull/1656/commits/02df5acc687bf04bea38c4033cf81513a20c7dcd